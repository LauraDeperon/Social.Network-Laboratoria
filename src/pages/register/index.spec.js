import { Register } from './index.js';
import * as service from '../../services/index.js';
import * as util from '../../utils/history.js';

service.signUp = jest.fn(() => Promise.resolve(true));
service.verifyEmail = jest.fn(() => Promise.resolve(true));
util.onNavigate = jest.fn(() => Promise.resolve(true));

describe('register', () => {
  it('should be a function', () => {
    expect(typeof Register).toBe('function');
  });
  it('should render the page', () => {
    expect(typeof Register).toMatchSnapshot();
  });
  // it('should verify the user password length', () => {
  //   Register().querySelector('#password').dispatchEvent(new Event('input'));
  //   return expect(verifyPasswordLength).resolves.toBe(true);
  // });
  // it('should confirm the user password', () => {
  //   Register().querySelector('#password').dispatchEvent(new Event('input'));
  //   return expect(verifyConfirmPassword).resolves.toBe(true);
  // });
  it('should call signUp when the user click the button', async () => {
    Register().querySelector('#signup-button').dispatchEvent(new Event('click'));
    await expect(service.signUp).toHaveBeenCalled();
    expect(service.verifyEmail).toHaveBeenCalled();
    expect(util.onNavigate).toHaveBeenCalledWith('/profile');
  });
});
