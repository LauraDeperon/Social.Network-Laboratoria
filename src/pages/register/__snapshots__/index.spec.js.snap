// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register should render the page 1`] = `"function"`;
